### ISSUE TITLE
lint unnecessary `.into_iter()`

### ISSUE THREAD

**Issue by zackmdavis** [Manishearth/rust-clippy#1094]
`for` loops [implicitly call `.into_iter()`](https://doc.rust-lang.org/std/iter/index.html#for-loops-and-intoiterator) on that which is being iterated over, but if someone didn't know this, they might have written `.into_iter()` themselves ([this is not merely a hypothetical scenario](https://github.com/zackmdavis/Leafline/commit/a4f59709ac6)), which is surely bad style.

Example of code that should get a warning (but does not as of Clippy v. 0.0.79):
```rust
fn main() {
    let my_vec = vec![1, 2, 3, 4];
    for i in my_vec.into_iter() {
        println!("{}", i);
    }
}
```
(could be just `for i in my_vec {`)

I'm guessing the implementation would probably be pretty similar to [that of the](https://github.com/Manishearth/rust-clippy/blob/f1a0ddf31b15d/clippy_lints/src/loops.rs#L497-L511) existing `explicit_iter_loop` lint?
